compiler/quilt/test/test_command.py

```diff
@@ -110,6 +110,36 @@ def test_version_add_declined(self, mock_input):
         mock_input.return_value = 'n'
         command.version_add('user/test', '2.9.12', 'fabc123')  # should produce no mock network activity
 
+    def test_ambiguous_hash(self):
+        registry_url = command.get_registry_url()
+        session = command._get_session()
+        ambiguous_token = "795a7b"
+        # There should be at least two results that start with the ambiguous_token, plus some non-ambiguous
+        # results in fake_data to test against.
+        fake_data = {'logs': [
+            {'author': 'user', 'created': 1490816524.0,
+             'hash': '885696c6e40613b3c601e95037caf4e43bda58c39f67ab5d5e56beefb3662ff4'},
+            {'author': 'user', 'created': 1490816507.0,
+             'hash': '795a7bc9e40613b3c601e95037caf4e43bda58c39f67ab5d5e56beefb3662ff4'},
+            {'author': 'user', 'created': 1490816473.0,
+             'hash': '795a7bc6e40613b3c601e95037caf4e43bda58c39f67ab5d5e56beefb3662ff4'},
+            {'author': 'user', 'created': 1490816524.0,
+             'hash': '2501a6c6e40a7b355901fc5037caf4e43bda58c39f67ab5d5e56beefb3662ff4'},
+        ]}
+        self.requests_mock.add(
+            responses.GET,
+            registry_url + "/api/log/user/test/",
+            json=fake_data
+        )
+        # Ambiguous hashes in _match_hash's exception will be sorted -- sorted here to match.
+        fake_data_ambiguous = sorted(entry['hash'] for entry in fake_data['logs']
+                               if entry['hash'].startswith(ambiguous_token))
+        # this will match each ambiguous hash, in order, separated by anything.
+        # ..it allows for formatting changes in the error, but requires the same order.
+        fake_data_regexp = '[\s\S]+'.join(fake_data_ambiguous)
```

Comment:
I'd use something like `(.|\n)` if that works. Otherwise, I guess that's fine. Also, in any case, add an `r` in front of the regex to make sure backslashes aren't interpreted the usual way: `r'(.|\n)'`

Reply:
Sure, it's just a common idiom.  `(.|\n)` looks nicer, though.  '\s\S' and r'\s\S' are identical, but the raw string is better in case anyone changes it at some point and adds a control character in there.